mpt-witness-generator/rustlib/src/lib.rs

```diff
@@ -0,0 +1,39 @@
+use std::{
+    ffi::{CStr, CString},
+    os::raw::c_char,
+};
+
+mod witness {
+    use super::*;
+    extern "C" {
+        pub fn GetWitness(str: *const c_char) -> *const c_char;
+    }
+}
+
+pub fn get_witness(data: &str) -> String {
+    let c_config = CString::new(data).expect("invalid config");
+    let result = unsafe { witness::GetWitness(c_config.as_ptr()) };
+    let c_str = unsafe { CStr::from_ptr(result) };
+    let c_str_string = c_str.to_str().expect("Error translating from library");
+    c_str_string.to_string()
```

Comment:
If my understanding is correct, the CStr is a type that holds a reference to an existing string, so this code doesn't own it and so Rust will not free it.  This string was allocated by go, but it will not be freed, leading a memory leak.
You can see how we resolved this in `geth-utils`: https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/1bc1eef6344fba33105b018216d7af5f04550cd8/geth-utils/src/lib.rs#L31

Feel free to move this to a new issue if you'd like to fix this in a new PR :)

Reply:
But shouldn't be fixed also here? We will continue the development of the generator here, right?